torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -411,6 +413,7 @@ void RequestCallbackImpl::processPythonRRefFetchCall(
       Message m =
           PythonRRefFetchRet(std::move(*result).toIValues()).toMessage();
       m.setId(messageId);
+      rref->waitAllDevices();
```

Comment:
we shouldn't call `waitAllDevices` in this way, because it currently only blocks current streams, but we don't haven't set up streams using the ctx, which means that after doing this, the streams are still not synchronized.